File: objective-c/test/Ice/objects/TestIntfI.m
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

#import <objc/Ice.h>
#import <objects/TestI.h>
#import <ObjectsDerived.h>
#import <ObjectsDerivedEx.h>

@implementation TestObjectsTestIntfI
-(TestObjectsBase*) opDerived:(ICECurrent *)__unused current
{
   TestObjectsDerived *d = ICE_AUTORELEASE([[TestObjectsDerived alloc] init]);
   d.theS.str = @"S.str";
   d.str = @"str";
   d.b = @"b";
   return d;
}

-(void) throwDerived:(ICECurrent *)__unused current
{
   @throw [TestObjectsDerivedEx derivedEx:@"reason"];
}
@end

File: cpp/test/Ice/objects/TestIntfI.cpp
// Copyright (c) ZeroC, Inc.

#include "Derived.h"
#include "DerivedEx.h"
#include "Ice/Ice.h"
#include "TestI.h"

using namespace Test;

BasePtr
TestIntfI::opDerived(const Ice::Current&)
{
    DerivedPtr d = std::make_shared<Derived>();
    d->theS.str = "S.str";
    d->str = "str";
    d->b = "b";
    return d;
}

void
TestIntfI::throwDerived(const Ice::Current&)
{
    throw DerivedEx{"reason"};
}
